diff -uNr a/src/modules/rlm_python3/rlm_python3.c b/src/modules/rlm_python3/rlm_python3.c
--- a/src/modules/rlm_python3/rlm_python3.c	2021-10-07 21:46:04.000000000 +0200
+++ b/src/modules/rlm_python3/rlm_python3.c	2022-05-18 21:13:21.802170096 +0200
@@ -204,12 +204,12 @@
 		return;
 	}

-	if (((pStr1 = PyObject_Str(pExcType)) != NULL) &&
+	if (((pStr1 = PyObject_Str(pExcType)) != NULL) &&
 	    ((pStr2 = PyObject_Str(pExcValue)) != NULL)) {
 		ERROR("%s:%d, Exception type: %s, Exception value: %s", __func__, __LINE__, PyUnicode_AsUTF8(pStr1), PyUnicode_AsUTF8(pStr2));
 		Py_DECREF(pStr1);
 		Py_DECREF(pStr2);
-	}
+	}

 	if (pExcTraceback) {
 		PyObject *pRepr = PyObject_Repr(pExcTraceback);
@@ -388,7 +388,7 @@
 		if (PyErr_Occurred()) {
 			python_error_log();
 		}
-
+
 		return -1;
 	}
